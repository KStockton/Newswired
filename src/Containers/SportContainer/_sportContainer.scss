.Sports-header {
  box-shadow: 2px 2px 2px #cab6b6;
  font-family: 'Noto Sans', sans-serif;
  font-size: 1.7rem;
  padding: 36px 0px;
}

.sports-wrapper {
  display: flex;
  overflow: scroll;
  width: 100vw;
  display: flex;
  overflow: scroll;
  width: 100vw;
  background-image: linear-gradient( rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.5) ),url('../Assets/sports.jpg'); 
  background-repeat: no-repeat;
  background-position-x: center;
  background-position-y: center;
  background-size: cover;
  }
  