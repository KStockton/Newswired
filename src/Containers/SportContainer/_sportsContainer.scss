.sports-wrapper {
  display: flex;
  overflow: scroll;
  width: 100vw;
  }
  