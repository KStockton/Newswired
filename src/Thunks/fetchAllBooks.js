import * as action from '../actions/index'



export const fetchAllBooks = (url) => {

  return async (dispatch) => {
    try {
      dispatch(action.isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      const books = await response.json()
      dispatch(action.isLoading(false));
      dispatch(action.getAllBooks(books))
    } catch (error) {
      dispatch(action.hasErrored(error.message))
    }
  }
}