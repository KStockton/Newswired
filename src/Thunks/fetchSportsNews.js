import * as action from '../actions/index';
import { cleanResponse } from './../Utility/Cleaners/cleanResponse'


export const fetchSportsNews = (url) => {

  return async (dispatch) => {
    try {
      dispatch(action.isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
      throw Error(response.sendText)
    }
      const data = await response.json()
      const cleanData = cleanResponse(data)
      dispatch(action.getSportsNews(cleanData))
      dispatch(action.isLoading(false))
    } catch(error) {
      dispatch(action.hasErrored(error.message, 'sportsNews'))
    }
  }
}