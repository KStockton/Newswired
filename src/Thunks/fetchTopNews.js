import * as action from '../actions/index';


export const fetchTopNews = (url) => {

  return async (dispatch) => {
    try {
      dispatch(action.isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
      throw Error(response.sendText)
    }
      const data = await response.json()
      dispatch(action.getTopNews(data))
      dispatch(action.isLoading(false))
    } catch(error) {
      dispatch(action.hasErrored(error.message, 'topNews'))
    }
  }
}

