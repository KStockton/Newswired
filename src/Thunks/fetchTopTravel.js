import * as action from '../actions/index';


export const fetchTopTravel = (url) => {
console.log('url', url)
  return async (dispatch) => {
    try {
      dispatch(action.isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
      throw Error(response.sendText)
    }
      const data = await response.json()
      console.log('data', data)
      dispatch(action.getTopTravel(data))
      dispatch(action.isLoading(false))
    } catch(error) {
      dispatch(action.hasErrored(error.message))
    }
  }
}

