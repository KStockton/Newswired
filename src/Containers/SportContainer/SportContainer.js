import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchSportsNews } from '../../Thunks/fetchSportsNews';
import Travel from '../../Components/Travel/Travel';
const shortid = require('shortid')
const API_KEY =`${process.env.REACT_APP_NEWSAPI_API_KEY}`
class SportsContainer extends Component {

   componentDidMount() {
    if(this.props.allSports.length == 0){
      const sportNewsUrl = 
      `https://newsapi.org/v2/everything?q=sports&language=en&from=2019-06-03&page=1&domains=espn.com&apiKey=${API_KEY}`;
      this.props.fetchSportsNews(sportNewsUrl)
    }
  }

  displaySportNews = () => {
   return this.props.allSports.map(location => {
    return (<Travel {...location} key={shortid.generate()} id={shortid.generate()}/>)
   })
   }
  
  render() {
    const displaySportNews = this.displaySportNews()
   
    return (
      <section>
      <h2>Sport News</h2>
      <div className="SportNews-wrapper">
        {displaySportNews}
      </div>
      </section>
    )
  }
}
export const mapStateToProps = (state) => ({
  allSports: state.allSports
})

export const mapDispatchToProps = (dispatch) => ({
  fetchSportsNews: (url) => dispatch(fetchSportsNews(url))
})

export default connect(mapStateToProps, mapDispatchToProps)(SportsContainer)