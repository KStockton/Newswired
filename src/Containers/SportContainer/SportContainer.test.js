import React from 'react'
import { SportContainer, mapStateToProps, mapDispatchToProps} from './SportContainer'
import { shallow } from 'enzyme';
import { fetchSportsNews } from '../../Thunks/fetchSportsNews'
jest.mock('../../Thunks/fetchAllBooks')

describe('SportsContainer',() => {
  let wrapper;
  let mockSports = { allSports :[{title:'Golden'},{title:'Xavier Basketball'}]}
  let mockError = 'something went wrong';
  let mockfetchSportNews = jest.fn()
  beforeEach(() => {
    wrapper = shallow(<SportContainer 
      allSports={mockSports}
      error={mockError}
      fetchSportsNews={mockfetchSportNews}
      />)
  })
  xit('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

})
describe('MSTP',() =>{
  it('should map state from mockstate', () => {
    const mockState = { allSports :[{title:'Golden'},{title:'Xavier Basketball'}], error: 'oops'}
    const result = mapStateToProps(mockState)
    expect(result).toEqual(mockState)
  }) 
});
describe('MDTP',() =>{
  const mockDispatch = jest.fn()
  const mappedProps = mapDispatchToProps(mockDispatch)
  xit('should dispatch action when correct params passed', () => {
    const mockUrl = 'www.cool.com'
    const actionToDispatch = fetchSportsNews(mockUrl)
    mappedProps.fetchSportsNews(mockUrl)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  }) 
});