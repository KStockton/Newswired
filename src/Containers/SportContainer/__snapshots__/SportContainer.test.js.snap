// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SportsContainer should match the snapshot 1`] = `
<section>
  <h2
    className="Sports-header"
  >
    Sport News
  </h2>
  <div
    className="sports-wrapper Card-wrapper"
  >
    <Card
      key="G72Ngd3k0"
      title="Golden"
    />
    <Card
      key="W21f07gZLd"
      title="Xavier Basketball"
    />
  </div>
</section>
`;

exports[`SportsContainer should render error message if the error is not an empty string and allsports.length === 0 1`] = `
<section>
  <h2
    className="Sports-header"
  >
    Sport News
  </h2>
  <div
    className="sports-wrapper Card-wrapper"
  >
    <div>
      <h2>
        something went wrong
      </h2>
      <Link
        to="/Options"
      >
        Back
      </Link>
    </div>
  </div>
</section>
`;

exports[`SportsContainer should render loading if the component is loading 1`] = `
<section>
  <h2
    className="Sports-header"
  >
    Sport News
  </h2>
  <div
    className="sports-wrapper Card-wrapper"
  >
    <Loading />
  </div>
</section>
`;
